cpp/src/stream_compaction/drop_duplicates.cu

```diff
@@ -130,18 +174,93 @@ std::unique_ptr<table> drop_duplicates(table_view const& input,
                         mr);
 }
 
+std::unique_ptr<table> unordered_drop_duplicates(table_view const& input,
+                                                 std::vector<size_type> const& keys,
+                                                 null_equality nulls_equal,
+                                                 rmm::cuda_stream_view stream,
+                                                 rmm::mr::device_memory_resource* mr)
+{
+  if (0 == input.num_rows() || 0 == input.num_columns() || 0 == keys.size()) {
+    return empty_like(input);
+  }
+
+  auto keys_view = input.select(keys);
+  auto table_ptr = cudf::table_device_view::create(keys_view, stream);
+  auto has_null  = nullate::DYNAMIC{cudf::has_nulls(keys_view)};
+  auto const num_rows{table_ptr->num_rows()};
+
+  hash_map_type key_map{compute_hash_table_size(num_rows),
+                        COMPACTION_EMPTY_KEY_SENTINEL,
+                        COMPACTION_EMPTY_VALUE_SENTINEL,
+                        detail::hash_table_allocator_type{default_allocator<char>{}, stream},
+                        stream.value()};
+
+  compaction_hash hash_key{has_null, *table_ptr};
+  row_equality_comparator row_equal(has_null, *table_ptr, *table_ptr, nulls_equal);
+
+  auto iter = cudf::detail::make_counting_transform_iterator(
+    0, [] __device__(size_type i) { return cuco::make_pair(std::move(i), std::move(i)); });
+  // insert unique indices into the map.
+  key_map.insert(iter, iter + num_rows, hash_key, row_equal, stream.value());
+
+  auto counting_iter = thrust::make_counting_iterator<size_type>(0);
+  rmm::device_uvector<bool> existences(num_rows, stream, mr);
```

Comment:
The name `existences` sounds strange to me, and does not clearly indicate what exists. It also does not suggest that the type is `bool`, because it could be an integer count of things that exist. Maybe `index_exists_in_map` instead?
```suggestion
  rmm::device_uvector<bool> index_exists_in_map(num_rows, stream, mr);
```